lib/worker.js

```diff
@@ -2,10 +2,44 @@
 
 const net = require('net');
 const JSON = require('./json-buffer');
+const fs = require('fs');
+const os = require('os');
+const path = require("path");
+const READFILETIMELIMIT = 5000;
+const debugMode = false;
+
+const responseFolder = path.join(os.tmpdir(), 'sync_rpc_responses');
+if (!fs.existsSync(responseFolder)) {
+  fs.mkdirSync(responseFolder, {recursive: true});
+}
+
+const responseFile = path.join(responseFolder, `responseFile.txt`);
+
+function responseFileReadyForWriting() {
+  return !fs.existsSync(responseFile);
+}
+
+function timeLimitExceeded(startTime) {
+  return startTime + READFILETIMELIMIT < Date.now();
+}
+
+function writeResponseToFile(response) {
+  const startTime = Date.now();
+  while (!responseFileReadyForWriting()) {
+    // if (timeLimitExceeded(startTime)) {
+    //   fs.unlinkSync(responseFile);
+    // }
+  }
+  fs.writeFileSync(responseFile, response);
+  debugMode && console.log(`Written response to file.`);
+  fs.chmodSync(responseFile, '444');
+  debugMode && console.log(`Changed response file to readonly.`);
+}
 
 const INIT = 1;
 const CALL = 0;
 const modules = [];
+const index = [0];
```

Comment:
why does this need to be an array?

Reply:
want to use reference so that we can increment the index on the requests.